query/sql_parser.go

```diff
@@ -354,9 +354,11 @@ func (v *ASTBuilder) VisitQueryNoWith(ctx *antlrgen.QueryNoWithContext) interfac
 	v.SQL2AqlCtx.exprOrigin = ExprOriginOthers
 	var orderBy = v.getOrderBy(ctx)
 
-	limit, err := strconv.Atoi(v.GetTextIfPresent(ctx.GetLimit()))
-	if err != nil {
-		v.Logger.Panic("failed to parse limit")
+	limit := 0
+	if limitParsed, err := strconv.Atoi(v.GetTextIfPresent(ctx.GetLimit())); err != nil {
+		v.Logger.Warn("failed to parse limit")
```

Comment:
this should be a parse error and return

Reply:
yeah.. I've updated the panic msg to contain more context. the panic will be recovered and throw to handler as error.